### PR TITLE
Port over code from puppet-foreman

### PR DESCRIPTION
**Pull request by ekohl**
These files were part of puppet-foreman but now become a standalone module. This makes it easier to test.

Various fixes were made to get tests to pass properly in the new layout and with the additional operating systems. No actual code was changed.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      "version_requirement": ">= 5.5.8 < 7.0.0"
+    }
+  ],
+  "operatingsystem_support": [
+    {
+      "operatingsystem": "RedHat",
+      "operatingsystemrelease": [
+        "6",
+        "7",
+	"8"
```
Comment (mmoll) [inline metadata.json:35]: tab indent.

File: metadata.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      ]
+    },
+    {
+      "operatingsystem": "CentOS",
+      "operatingsystemrelease": [
+        "6",
+        "7",
+	"8"
```
Comment (mmoll) [inline metadata.json:43]: tab indent

**Comment by mmoll** [comment]
Could travis get activated for this repo?

**Comment by ekohl** [comment]
Done. Now it fails because it needs facterdb 1.3.0: https://github.com/voxpupuli/voxpupuli-test/commit/0c275343aa748a1d00ec66ca0e4aac443166ecf4

**Comment by ekohl** [comment]
Argh, only tested with Puppet 6 and that was green. Will get to this later.

**Comment by mmoll** [comment]
Also needs a rebase now :)

**Comment by ekohl** [comment]
Due to time constraints I'll get back to this once Foreman 2.1 is branched

**Comment by ekohl** [comment]
Rebased and updated

**Pull request merged**